Errors to fix:

external/libsdl12/src/video/Xext/Xinerama/Xinerama.c:61:5: error: initialization of 'int (*)(Display *, XExtCodes *)' from incompatible pointer type 'int (*)(void)' [-Wincompatible-pointer-types]

or:

In file included from external/libsdl12/src/video/Xext/XME/xme.c:27:
external/libsdl12/src/video/Xext/XME/xme.c:226:37: error: conflicting types for 'XiGMiscCloseDisplay'; have 'int(Display *, XExtCodes *)'
  226 | static XEXT_GENERATE_CLOSE_DISPLAY (XiGMiscCloseDisplay, xigmisc_info)
      |                                     ^~~~~~~~~~~~~~~~~~~
external/libsdl12/src/video/Xext/XME/../extensions/extutil.h:203:5: note: in definition of macro 'XEXT_GENERATE_CLOSE_DISPLAY'
  203 | int proc (Display *dpy, XExtCodes *codes) \
      |     ^~~~
external/libsdl12/src/video/Xext/XME/xme.c:209:12: note: previous declaration of 'XiGMiscCloseDisplay' with type 'int(void)'
  209 | static int XiGMiscCloseDisplay();
      |            ^~~~~~~~~~~~~~~~~~~
AI Mode reply for In file included from external/libsdl12/src/video/Xext/XME/xme.c:27:
external/libsdl12/src/video/Xext/XME/xme.c:226:37: error: conflicting types for 'XiGMiscCloseDisplay'; have 'int(Display *, XExtCodes *)'
  226 | static XEXT_GENERATE_CLOSE_DISPLAY (XiGMiscCloseDisplay, xigmisc_info)
      |                                     ^~~~~~~~~~~~~~~~~~~
external/libsdl12/src/video/Xext/XME/../extensions/extutil.h:203:5: note: in definition of macro 'XEXT_GENERATE_CLOSE_DISPLAY'
  203 | int proc (Display *dpy, XExtCodes *codes) \
      |     ^~~~
external/libsdl12/src/video/Xext/XME/xme.c:209:12: note: previous declaration of 'XiGMiscCloseDisplay' with type 'int(void)'
  209 | static int XiGMiscCloseDisplay();
      |            ^~~~~~~~~~~~~~~~~~~

diff --git a/src/video/Xext/Xinerama/Xinerama.c b/src/video/Xext/Xinerama/Xinerama.c
index 0000000..1111111
--- a/src/video/Xext/Xinerama/Xinerama.c
+++ b/src/video/Xext/Xinerama/Xinerama.c
@@ -51,5 +51,5 @@
 #define PanoramiXSimpleCheckExtension(dpy,i) \
   XextSimpleCheckExtension (dpy, i, panoramiX_extension_name)
 
-static int close_display();
+static int close_display(Display *dpy, XExtCodes *codes);
 static /* const */ XExtensionHooks panoramiX_extension_hooks = {
diff --git a/src/video/Xext/XME/xme.c b/src/video/Xext/XME/xme.c
index 0000000..2222222
--- a/src/video/Xext/XME/xme.c
+++ b/src/video/Xext/XME/xme.c
@@ -207,5 +207,5 @@
  * find_display - locate the display info block
  */
-static int XiGMiscCloseDisplay();
+static int XiGMiscCloseDisplay(Display *dpy, XExtCodes *codes);
 
 static XExtensionHooks xigmisc_extension_hooks = {
diff --git a/src/video/Xext/Xv/Xv.c b/src/video/Xext/Xv/Xv.c
index 0000000..3333333
--- b/src/video/Xext/Xv/Xv.c
+++ b/src/video/Xext/Xv/Xv.c
@@ -63,7 +63,7 @@
 #define XvCheckExtension(dpy, i, val) \
   XextCheckExtension(dpy, i, xv_extension_name, val)
 
-static char *xv_error_string();
-static int xv_close_display();
-static Bool xv_wire_to_event();
+static char *xv_error_string(Display *dpy, int code, XExtCodes *codes, char *buf, int n);
+static int xv_close_display(Display *dpy, XExtCodes *codes);
+static Bool xv_wire_to_event(Display *dpy, XEvent *host, xEvent *wire);
 
 static XExtensionHooks xv_extension_hooks = {
